docs/notebooks/multi_objective_ehvi.pct.py

```diff
@@ -0,0 +1,219 @@
+
+np.random.seed(1793)
+tf.random.set_seed(1793)
+
+# %% [markdown]
+# ## Describe the problem
+#
+# In this tutorial, we provide multi-objective optimization example using the expected hypervolume improvement acquisition function.
+# We consider the VLMOP2 function --- a synthetic benchmark problem with two objectives. We start by defining the problem parameters.
+
+
+# %%
+vlmop2 = VLMOP2().objective()
+observer = trieste.utils.objectives.mk_observer(vlmop2, OBJECTIVE)
+
+# %%
+mins = [-2, -2]
+maxs = [2, 2]
+search_space = Box(mins, maxs)
+num_objective = 2
+
+# %% [markdown]
+# Let's randomly sample some initial data from the observer ...
+
+# %%
+num_initial_points = 10
+initial_query_points = search_space.sample(num_initial_points)
+initial_data = observer(initial_query_points)
+
+# %% [markdown]
+# ... and visualise the data across the design space.
+
+# %%
+_, ax = plot_function_2d(
+    vlmop2,
+    mins,
+    maxs,
+    grid_density=100,
+    contour=True,
+    title=["Obj 1", "Obj 2"],
+    figsize=(12, 6),
+    colorbar=True,
+    xlabel="$X_1$",
+    ylabel="$X_2$",
+)
+plot_bo_points(initial_query_points, ax=ax[0, 0], num_init=num_initial_points)
+plot_bo_points(initial_query_points, ax=ax[0, 1], num_init=num_initial_points)
+plt.show()
+
+# %% [markdown]
+# ... and in the objective space
+
+# %%
+plot_mobo_points_in_obj_space(initial_data[OBJECTIVE].observations)
+plt.show()
+
+
+# %% [markdown]
+# ## Modelling the two functions
+#
+# In this example we model the two objective functions individually with their own Gaussian process models. For problems where the objective functions are similar it may make sense to build a joint model rather than a model stack.
+
+
+# %%
+def create_single_outcome_model(data, outcome_idx, input_dim=2, l=1.0):
+    variance = tf.math.reduce_variance(tf.gather(data.observations, [outcome_idx], axis=1))
+    lengthscale = l * tf.ones(input_dim, dtype=gpflow.default_float())
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=lengthscale)
+    single_outcome_dataset = Dataset(
+        data.query_points, tf.gather(data.observations, [outcome_idx], axis=1)
+    )
+    gpr = gpflow.models.GPR(single_outcome_dataset.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+    return create_model(
+        {
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+                "minimize_args": {"options": dict(maxiter=100)},
+            },
+        }
+    )
+
+
+# %%
+objective_models = [
+    (
+        create_single_outcome_model(
+            initial_data[OBJECTIVE],
+            outcome_idx=i,
+        ),
+        1,
+    )
+    for i in range(num_objective)
+]
+
+# %% [markdown]
+# Stack the two independent GP model in a single `ModelStack` multi-output model.
+
+# %%
+models = {OBJECTIVE: ModelStack(*objective_models)}
+
+# %% [markdown]
```

Comment:
This whole part could be done in a single function e.g. 

`def build_stacked_independent_objectives_model(data: Dataset) -> GaussianProcessRegression:
        gprs =[]
        for idx in range(2):
            single_obj_data = Dataset(data.query_points, tf.gather(data.observations, [idx], axis=1))
            variance = tf.math.reduce_variance(single_obj_data.observations)
            kernel = gpflow.kernels.Matern52(variance, tf.constant([0.2, 0.2], tf.float64))
            gpr = gpflow.models.GPR((single_obj_data.query_points, single_obj_data.observations), kernel, noise_variance=1e-5)
            gpflow.utilities.set_trainable(gpr.likelihood, False)
            gprs.append(GaussianProcessRegression(gpr))

        return ModelStack(*gprs)
`

Reply:
Something like that anyway